### ISSUE TITLE
nixos-install -I is broken

### ISSUE THREAD

**Issue by yegortimoshenko** [NixOS/nixpkgs#29986]
### Steps to reproduce

```
$ cat /mnt/etc/nixos/configuration.nix
{ imports = [ <overlay> ]; }
$ nixos-install -I overlay=/tmp/overlay
error: file ‘overlay’ was not found in the Nix search path (add it using $NIX_PATH or -I)
```

## Technical details

* System: 18.03pre116572.9824ca6975 (Impala)
* Nix version: nix-env (Nix) 1.11.15
* Nixpkgs version: "18.03pre116572.9824ca6975"
* Sandboxing enabled: false

**Comment by bjornfor** [comment]
Wild guess: nixos-install chroots to /mnt first, so the overlay would have to be in /mnt/tmp/overlay for it to work?

**Comment by yegortimoshenko** [comment]
Perhaps (but not documented, I'll try this out when I get the chance). But my bigger concern is that it's reasonable to assume that if I add a channel before `nixos-install` via `nix-channel --add https://github.com/foo/overlay/archive/master.tar.gz overlay` it'll be used in the installation process, but it's not.

**Comment by danbst** [comment]
No, it's because `nixos-install` doesn't pass it's arguments to underlying `nix-build`. Often, if you write a wrapper around `nix-build`, you don't expect users will see `nix-build` failures as your wrapper failures.

In this case it is not a bug, but feature request - pass `nixos-install` args to underlying `nix-build`

....

Ah, it's even worse then I thought

https://github.com/NixOS/nixpkgs/blob/e9f1d8693a666af293d690fabe561eea29c748a1/nixos/modules/installer/tools/nixos-install.sh#L31-L35

**Comment by yegortimoshenko** [comment]
From `nixos-install(1)`:

```
       -I
           Add a path to the Nix expression search path. This option may be given multiple times. See the NIX_PATH
           environment variable for information on the semantics of the Nix search path. Paths added through -I take
           precedence over NIX_PATH.
```

@danbst, why is it worse than you thought? It seems to do what it should do: add `-I` and its argument to build flags to be used in `nix-build` further down the road.

**Comment by danbst** [comment]
@yegortimoshenko oh man, you're right, I just can't read bash that fluently. So, we see it gets passed to `extraBuildFlags` array, which is used in `closure()` function

https://github.com/NixOS/nixpkgs/blob/e9f1d8693a666af293d690fabe561eea29c748a1/nixos/modules/installer/tools/nixos-install.sh#L121-L123

which is used in this actual `nix-build` call

https://github.com/NixOS/nixpkgs/blob/e9f1d8693a666af293d690fabe561eea29c748a1/nixos/modules/installer/tools/nixos-install.sh#L127-L131

As you see, it involves 2 calls to nix-build, one propagates build flags, the other misses those.

**Comment by danbst** [comment]
I'd cast @copumpkin here, as he knows his code best

**Comment by copumpkin** [comment]
It's quite possible I broke it. I'll take a look in the next few days, sorry!

**Comment by yegortimoshenko** [comment]
@copumpkin :ghost:

**Comment by spacefrogg** [comment]
As a heads up, I fixed this in #34527 without knowing this issue existed.
